### DISCUSSION TITLE
Multiple proposition for the skin editor

### DISCUSSION THREAD

**Discussion by Avelia-tr** [ppy/osu#27439] [Ideas]
# changing the color 
At the moment everything is blocked to white wich is really annoying for little skinning like modifying the music select screen. I think having a way of choosing wich color everything is could be really cool. it could use the already existing interface for chosing the combo color in the map editor. 
mockup of what it could look like 
![image](https://github.com/ppy/osu/assets/154550569/24d94186-7476-47f6-9ef9-08b695f684aa)

Those could also be dynamic depending on a chosen thing but I imagine this is easier said than done

# supporting custom fonts for skinning elements and/or the entire game
At the moments we can choose between 4 fonts. I think allowing to add fonts to this list could allow to a greater potential to use the skinning element in the future since it'll be easier to make everything fits the esthetic the player want. 

# add new skin element 
some of those are probably a bit unrealistic but I'll put it here nonetheless

I would propose 4 main one wich would make skinning greatly advance : 

Beatmap specific things like : 
- beatmap tags 
- beatmap genre/language
- beatmap backgrounds

leaderboard specific thing : 
- player for x rank
- flag of the player 
- and so on... 

More player specific thing like : 
- player rank 
- player avg acc 
- and every other player stat
- current skin used

more pp/sr related thing : 
- a pp for % acc 
- separated skill sr value (like aim value, speed value, acc value, fl value) 

Those addition might need a redesign for the dropdown to avoid flooding 

# Custom button/deactivating some button

This one is probably the most unrealistic proposition from this post : 

I think that adding the ability for sprite to be button, that means having the ability to do things when clicked could be really neat. 
For example giving the ability to a sprite that when clicked it apply some mods or change the game modes or even switch the tabs of the beatmap display

deactivating button could be done by simply hiding the button with an object to avoid clicking on thing made invisible by the skin. 

# add better layering/clarifying thiings 

I know this one is probably already planned but I'm gonna add it nonetheless cuz I still have some issue with current state of things
Currently you can't place at other places than the default layer. but it makes a lot of thing a bit confusing since they seem to affect that but don't and just move the given element in the current layer. I don't have any proposition for that but it's really confusing when first approaching the skin editor. 
those 2 option are the main culprit I think 
![image](https://github.com/ppy/osu/assets/154550569/44d2931d-cfe1-4193-9d1e-5c4fc0a7a479)

# add ability to fuse/overlay the song select and gameplay part of two different skin

having to do some modification on all skins that a player use can be really tiresome. I for example use a different skin for high ar and I know that some people change skin daily. 
the most important thing of a skin is the gameplay, but skin that uses the full capacity of the skin editor are really rare (I never heard about it but they might exist for all I know) Coming back to the default song select could be a bit desorienting/tiresome. 

# Add score result skinning

I don't know why this isn't already the case (probably some complication in how it works). But the result score is the last main area (by that I mean something that all the player constently experiences), that has no kind of edition available. It would be really neat to see that be a thing. Although this is easier said than done. 

Some base propositon that I have include : 
- customizing what compose/the order of the extended result screen
- customizing the colors of the cards
- customizing the info directly available on the card 
- customizing the size of those info 

I'll understand if some of those proposition aren't realistic, I'm not skilled/knowable enough to be able to distinct them so I put them all 
That was all thx for reading this post.

**Comment by OwenCMYK** [comment]
I think many of these are good idea, however the "current skin used" is somewhat unnecessary because you can just put a text label with the name of the skin. By definition the text label will only be visible on that one skin so it would be functionally the same.

**Reply by Avelia-tr** [reply]
I was thinking of it alongside with the fusing/overlaying skin together. I should have mentionned it there. (I also didn't think about that lol)

**Reply by OwenCMYK** [reply]
It's funny that I forgot about the skin overlaying idea considering I was the one who suggested it, but regardless it'll still be tricky because with the skin overlays there would be multiple skins, so it would be difficult to know which skins should be considered the "currently used skin", and at that point you might as well just use a text label anyway.
